bin/cli.js

```diff
@@ -28,24 +28,24 @@ const undeployGoal = require('../lib/goals/undeploy');
 module.exports = async function run (options) {
   try {
     const config = await nodeshiftConfig(options);
-    let enrichedResources = {};
+    const response = {};
     if (options.cmd === 'resource' || options.cmd === 'deploy' || options.cmd === 'apply-resource') {
-      enrichedResources = await resourceGoal(config);
+      const enrichedResources = await resourceGoal(config);
+      response.resources = enrichedResources;
+      if (options.cmd === 'deploy' || options.cmd === 'apply-resource') {
```

Comment:
I guess it doesn't matter if you create the deployments and services and routes and stuff before you have a build/image stream.  I think the reason i had it in the order i did was probably me just copying maven.   so +1 :)

Reply:
Yeah, with this change everything we get back from OpenShift is returned, so that information is included in the [`appliedResources`](https://gist.github.com/lance/9bc9e7d974892ffb8597e6967dd95954#file-nodeshift-json-L151) property.

Reply:
was this comment suppose to be for https://github.com/bucharest-gold/nodeshift/pull/150#discussion_r155128217

Reply:
Uhh yes

Reply:
@lance any thoughts on the order in which the goals are executed?  Should the build stuff go first?  i think it really only matters for what the console output looks like

Reply:
@lholmquist have a look at the latest change. I have refactored cli.js a bit and I think it makes things clearer. It also adjusts the execution order a bit (which may make this a minor [or major] version bump). The results now look like this: https://gist.github.com/lance/cb7a0ca991712723d3d18c1484fb3fae

Reply:
@lance i think this looks pretty good.  I made another comment about a rogue console.error.  and it looks like there are some tests breaking.  but 👍